src/optimizer/utils.ts

```diff
@@ -102,6 +104,9 @@ export function resolveAliasFromURL(url: string, aliases: ExportableAlias[]): st
       alias = `${aliasPart}_${counterPart}`;
     }
 
+    // Replace unsupported characters, but allow $ at the start
+    alias = `$${alias.substring(1).replace(unsupportedAliasCharRegex, '-')}`;
```

Comment:
Wouldn't it be better to do this on line 98 above? E.g.
```ts
const aliasPart = `$${rawAlias.replace(unsupportedAliasCharRegex, '-')}`;
```
This avoids prepending `$` twice and needing to use `substring(1)`. More importantly, however, I think it works better w/ the existing alias chech on lines 101 - 105. We want to do that check using the final alias (post-replacements), not the pre-replacement alias.

Reply:
Oh.. yeah. I'm not sure why I didn't think of that earlier.

Reply:
Added in [9dcef55](https://github.com/FHIR/GoFSH/pull/254/commits/9dcef55564c7708af4d1a0690c721cb62b567329).